HELargeCenterTabBarController.h

```diff
@@ -46,7 +46,7 @@ If we need it that someone external needs to delegateness, then we can work that
 */
 @interface HELargeCenterTabBarController : UITabBarController  <UITabBarControllerDelegate>
 
-- (void)addCenterButtonWithUnselectedImage:(UIImage*)unselectedImage selectedImage:(UIImage*)selectedImage target:(id)target action:(SEL)action;
+- (void)addCenterButtonWithUnselectedImage:(UIImage*)unselectedImage selectedImage:(UIImage*)selectedImage target:(id)target action:(SEL)action allowSwitch:(BOOL)allowSwitch;
```

Comment:
Actually, could you change this so as to preserve the existing API?

So have 2 `-addCenter` methods, the old way and the new way.  Then in implementation, your new `allowSwitch` version of course contains all the heavy lifting. But the old compatibility API is implemented merely by calling the new method but passing a reasonable default for `allowSwitch` so as to preserve the existing API and behavior.